data-prepper-core/src/main/java/org/opensearch/dataprepper/DataPrepper.java

```diff
@@ -94,6 +98,7 @@ public boolean execute() {
 
     public void shutdown() {
         shutdownPipelines();
+        extensionsApplier.shutdownExtensions();
```

Comment:
We have an interface for `DataPrepperShutdownListener`. If we have `ExtensionApplier` register itself as a shutdown listener, then we won't have to change this class at all. Also, we can keep `ExtensionApplier` package protected.

An alternative approach would be to use Spring to shutdown and then have `ExtensionApplier` shutdown from that. 

Either way, we should stop adding new shutdown calls in here because it will be difficult to maintain.